model/client4.go

```diff
@@ -1748,7 +1755,18 @@ func (c *Client4) GetFile(fileId string) ([]byte, *Response) {
 	if r, err := c.DoApiGet(c.GetFileRoute(fileId), ""); err != nil {
 		return nil, BuildErrorResponse(r, err)
 	} else if data, err := ioutil.ReadAll(r.Body); err != nil {
-		return nil, &Response{StatusCode: r.StatusCode, Error: NewAppError("GetFile", "model.client.read_file.app_error", nil, err.Error(), r.StatusCode)}
+		return nil, BuildErrorResponse(r, NewAppError("GetFile", "model.client.read_file.app_error", nil, err.Error(), r.StatusCode))
+	} else {
+		return data, BuildResponse(r)
+	}
+}
+
+// DownloadFile gets the bytes for a file by id, optionally adding headers to force the browser to download it
+func (c *Client4) DownloadFile(fileId string, download bool) ([]byte, *Response) {
+	if r, err := c.DoApiGet(c.GetFileRoute(fileId)+fmt.Sprintf("?download=%v", download), ""); err != nil {
+		return nil, BuildErrorResponse(r, err)
+	} else if data, err := ioutil.ReadAll(r.Body); err != nil {
```

Comment:
I think there are a few places where we forget this: You should always close the `Body` field of an `http.Response`.